### ISSUE TITLE
Firefox Flatpak is able to make system unresponsible even with systemd-oomd enabled

### ISSUE THREAD

**Issue by MateusRodCosta** [systemd/systemd#30668]
### systemd version the issue has been seen with

systemd-254.7-1.fc39.x86_64

### Used distribution

Fedora Silverblue 39

### Linux kernel version used

6.6.8-200.fc39.x86_64

### CPU architectures issue was seen on

x86_64

### Component

systemd-oomd

### Expected behaviour you didn't see

I use a laptop with 16 GB, the biggest consumer of it usually is Firefox (as I very often have 300+ tabs open at once).
Other applications that usually consume a lot of RAM are Android Studio (such as Gradle during compilation or the Android emulator) and VMs using libvirt + QEMU (though I can choose how much RAM I want to be accessible to the VM).

Firefox is installed via Flatpak whereas Android Studio and libvirt and QEMU are installed on the host.

On scenarios where the system is running out of RAM, which is due to Firefox, Firefox is not killed.

### Unexpected behaviour you saw

Instead of Firefox being killed what usually happens is:

* System goes unresponsive for such long periods of time that I have no choice but to force shutdown
* Other applications get killed, such as Android Studio or the VMs (even though they are using a lot of RAM, it would make more sense to kill Firefox in those scenarios)

### Steps to reproduce the problem

0. Have a system with a "decent" amount of RAM
1. Install Firefox as flatpak
2. Develop an insane way of navigating the Internet that makes you easily accumulate more than 200 tabs at once
3. Try to use any other slightly memory-hungry application alongside the browser with 200+ tabs open
4. Just wait until the system slows down to a crawl due to running out of RAM
5. Notice how systemd-oomd won't save you

**Comment by septatrix** [comment]
Do you have swap enabled? (Fedora usually should ship Swap with ZRAM unless you disabled it - I expect the same to be the case for Silverblue)

**Comment by MateusRodCosta** [comment]
> Do you have swap enabled? (Fedora usually should ship Swap with ZRAM unless you disabled it - I expect the same to be the case for Silverblue)

Yes, I have zram enabled with basically the default options except with zstd compression:

```
$ zramctl 
NAME       ALGORITHM DISKSIZE  DATA COMPR TOTAL STREAMS MOUNTPOINT
/dev/zram0 zstd            8G  7,1M 25,7K  212K      12 [SWAP]
```

(I did also try a BTRFS swap file shortly but have since given up)

Even then I manage to exhaust RAM with Firefox.

What seems most likely in my system, and is my current working hypothesis, is that systemd-oomd for whatever reason works fine with non-Flatpak apps (such as killing Android Studio and QEMU when they use too much RAM), but it fails to detect Firefox is using too much RAM and doesn't kill it.

**IMPORTANT**: Btw, one very interesting thing, I remember that previously it used to be very common that systemd-oomd would kill Firefox, but that stopped a while ago, AFAIK the last time I remember systemd-oomd properly killing Firefox flatpak due to running out of memory was towards the end of H1 of 2023.

**Comment by bluca** [comment]
Have you reported this to Fedora? It seems like a configuration problem

**Comment by mikedlr** [comment]
There is a very longstanding Fedora bug that Firefox (not just inside a Flatpak) can lock the system up. 

https://bugzilla.redhat.com/show_bug.cgi?id=1597028
